### PR TITLE
Making `safetensors` a core dependency.

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

Making `safetensors` a core dependency.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by sgugger — COMMENTED** [review]
If it is in the `install_requires` it doesn't need to be anywhere else.

**Comment by Narsil** [comment]
> If it is in the `install_requires` it doesn't need to be anywhere else.

`tokenizers` is in the `install_requires` too, yet in a bunch of other places (I merely copied it). Isn't `tokenizers`  a core dependency ?

**Comment by sgugger** [comment]
Yes it is. It was not added by me and way before I was asked to review PRs for Transformers ;-)

**Review by amyeroberts — APPROVED** [review]
LGTM - thanks for adding! 

Happy to have as a core dependency, but let's get the opinions of other maintainers before merging :)

**Comment by Narsil** [comment]
Linked blogpost : https://github.com/huggingface/blog/pull/1096

**Comment by patrickvonplaten** [comment]
This means that weights are now always loaded in safetensors format but still saved in PyTorch format no? Think this is a good first step. Don't see a problem with having `safetensors` as a core dependency

File: setup.py
Comment (sgugger) [inline setup.py:276]:
```suggestion
extras["safetensors"] = deps_list("safetensors")
```
Like the one for tokenizers, this is really superfluous as you can't escape the packages in `install_requires`.

**Review by sgugger — APPROVED** [review]
Works for me! The CI already installs it since November, and this way we might catch more issues with safetensors loading.

**Review by LysandreJik — APPROVED** [review]

**Comment by Narsil** [comment]
> This means that weights are now always loaded in safetensors format but still saved in PyTorch format no? Think this is a good first step. Don't see a problem with having safetensors as a core dependency

Indeed. 

The next step will be saving in `safetensors` first. But we need to let time pass so that we can ensure a vast majority of users has `safetensors` so that users using a somewhat old transformers can  still load new models (finetuned versions of existing models in old versions)

**Comment by Narsil** [comment]
Merging !

**Pull request merged**
